Sources/SourceControl/Repository.swift

```diff
@@ -27,21 +27,23 @@ public struct RepositorySpecifier: Hashable, Sendable {
     }
 
     /// Create a specifier on a URL.
-    public init(url: URL) {
+    public init(url: GitURL) {
         self.init(location: .url(url))
     }
 
-    /// The location of the repository as URL.
-    public var url: URL {
+    /// The location of the repository as string.
+    public var url: String {
         switch self.location {
-        case .path(let path): return URL(fileURLWithPath: path.pathString)
-        case .url(let url): return url
+        case .path(let path): return path.pathString
+        case .url(let url): return url.absoluteString
         }
     }
 
     /// Returns the cleaned basename for the specifier.
     public var basename: String {
-        var basename = self.url.pathComponents.dropFirst(1).last(where: { !$0.isEmpty }) ?? ""
+        // FIXME: this might be wrong
+        //var basename = self.url.pathComponents.dropFirst(1).last(where: { !$0.isEmpty }) ?? ""
+        var basename = (self.url as NSString).lastPathComponent
```

Comment:
I don't fully understand what the previous code does here (why the `dropFirst(1)` when we're then doing `last`), so this might be wrong.

Reply:
Looks like the tests uncovered the purpose here, it is to make `basename` of "/" be the empty string.